scripts/zones/Bostaunieux_Oubliette/npcs/Novalmauge.lua

```diff
@@ -15,39 +15,38 @@ require("scripts/globals/quests")
 
 local path =
 {
-    41.169430, -24.000000, 19.860674,
-    42.256676, -24.000000, 19.885197,
-    41.168694, -24.000000, 19.904638,
-    21.859211, -24.010996, 19.792259,
-    51.917370, -23.924366, 19.970068,
-    74.570229, -24.024828, 20.103880,
-    44.533886, -23.947662, 19.926519
+    [1] = {20.000, -24.000, 20.000, 0},
+    [2] = {50.000, -24.000, 20.000, 0},
+    [3] = {75.000, -24.000, 20.000, 0},
+    [2] = {50.000, -24.000, 20.000, 0}
 }
 
 local wsQuest = tpz.wsquest.spiral_hell
 
 function onSpawn(npc)
     npc:initNpcAi()
-    npc:setPos(tpz.path.first(path))
+    npc:setPos(path[1][1], path[1][2], path[1][3], 0)
+    npc:speed(12)
+    npc:setPathPoint(1)
     onPath(npc)
 end
 
 function onPath(npc)
-    tpz.path.patrol(npc, path)
+    tpz.path.npcBasicPath(npc, path, 0)
 end
 
 function onTrade(player, npc, trade)
     local wsQuestEvent = tpz.wsquest.getTradeEvent(wsQuest, player, trade)
-
+    
     if player:getCharVar("troubleAtTheSluiceVar") == 2 and npcUtil.tradeHas(trade, 959) then -- Dahlia
         player:startEvent(17)
-        npc:wait()
+        npc:speed(0)
```

Comment:
If all paroling NPCs use this new method, will `npc:wait()` still have a use? Should we consider changing the internals of `:wait()` to essentially set speed to 0? (Would pathing need to be stopped?)

If we need to keep `npc:wait()`, should there be a wrapper like `tpz.path.pause()`?

Reply:
onion: wait() does a lot of things besides stopping movement, it stops all the stuff and things. because of that its tempting to use it for a lot of situations that result in crashes. wait should die form its use in pathing scripts, but may have future use for..something.. 

could we possibly cause the event triggering core side to set the entities speed to 0 and then the event finish to restore it to its original value? :thinking: `m_event.Target` appears to be the npc in the core.

Reply:
> m_event.Target appears to be the npc in the core

I was not aware of this - we can make event starts accurate! (On retail, the packet is the initiating NPC, but we've always invoked it directly as the "player" being the initiator)

Reply:
I'm not really sure where to go with this but as we now have pathStop() and pathResume(), we will no longer get crashes when interacting with an npc that uses pathing, i think i found and converted all npc's that use pathing to use the new pathing systems.

Reply:
I can look into making event set the speed to 0 at a later date if still wanted.